dhis2/src/test/java/org/motechproject/dhis2/tasks/builder/ChannelRequestBuilderTest.java

```diff
@@ -182,85 +159,61 @@ public void testActionEventRequests() throws Exception{
-        program1.setName("Program1");
-        program1.setUuid("program1UUID");
-        program1.setRegistration(true);
-        program1.setSingleEvent(false);
-        program1.setTrackedEntity(trackedEntity);
-        program1.setStages(stages);
-        program1.setAttributes(attributes);
-
-        Program program2 = new Program();
-        program2.setName("Program2");
-        program2.setUuid("program2UUID");
-        program2.setRegistration(true);
-        program2.setSingleEvent(false);
-        program2.setTrackedEntity(trackedEntity);
-        program2.setStages(stages);
-        program2.setAttributes(attributes);
+        when(programService.findByRegistration(true)).thenReturn(programs);
+        when(stageService.findAll()).thenReturn(stages);
+        when(trackedEntityAttributeService.findAll()).thenReturn(attributes);
+        when(trackedEntityService.findAll()).thenReturn(trackedEntities);
+        when(dataSetService.findAll()).thenReturn(dataSets);
 
-        programs.add(program1);
-        programs.add(program2);
+        when(bundleContext.getBundle()).thenReturn(bundle);
+        when(bundle.getVersion()).thenReturn(version);
+        when(dhisWebService.getServerVersion()).thenReturn(serverVersion);
 
+        when(bundle.getSymbolicName()).thenReturn("BundleSymbolicName");
+        when(version.toString()).thenReturn("bundleVersion");
     }
 
-    private void setupStages() {
-
-        DataElement dataElement1 = new DataElement("dataElementName1","dataElementID1");
-        DataElement dataElement2 = new DataElement("dataElementName2","dataElementID2");
-
-        Stage stage1 = new Stage();
-        stage1.setName("stage1");
-        stage1.setProgram("programID");
-        stage1.setUuid("stageID");
-
-        List<DataElement> dataElements = new ArrayList<>();
-        dataElements.add(dataElement1);
-        dataElements.add(dataElement2);
-
-        stage1.setDataElements(dataElements);
-
-        Stage stage2 = new Stage();
-        stage2.setName("stage2");
-        stage2.setProgram("programID");
-        stage2.setUuid("stageID");
+    private void mockActionEventRequest() {
+        when(stageActionBuilder.build(stages)).thenReturn(prepareStagesActionEventRequests());
+        when(createInstanceActionBuilder.build(attributes, trackedEntities)).thenReturn(prepareInstanceActionEventRequests());
+        when(programActionBuilder.build(programs, serverVersion)).thenReturn(prepareProgramActionEventRequests());
+        when(sendDataValueSetActionBuilder.addSendDataValueSetActions(dataSets)).thenReturn(prepareDataSetsActionEventRequests());
+    }
 
-        stage2.setDataElements(dataElements);
+    private List<ActionEventRequest> prepareProgramActionEventRequests() {
+        ProgramActionBuilder builder = new ProgramActionBuilder();
+        List<ActionEventRequest> result = builder.build(programs, serverVersion);
 
-        stages.add(stage1);
-        stages.add(stage2);
+        return result;
```

Comment:
return builder.build(programs, serverVersion);

or do it whole method login in one line